SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -380,3 +384,68 @@ void ULayeredLBStrategy::AddStrategyForLayer(const FName& LayerName, UAbstractLB
 	LayerNameToLBStrategy.Add(LayerName, LBStrategy);
 	LayerNameToLBStrategy[LayerName]->Init();
 }
+
+TUniquePtr<SpatialGDK::FLoadBalancingCalculator> ULayeredLBStrategy::CreateLoadBalancingCalculator(FLegacyLBContext& OutCtx) const
+{
+	if (!IsStrategyWorkerAware())
+	{
+		return nullptr;
+	}
+
+	TArray<FName> LayerNames;
+	TArray<TUniquePtr<SpatialGDK::FLoadBalancingCalculator>> Layers;
+	LayerNames.SetNum(LayerNameToLBStrategy.Num());
+	Layers.SetNum(LayerNameToLBStrategy.Num());
+
+	for (auto Entry : LayerNameToLBStrategy)
+	{
+		FName LayerName = Entry.Key;
+		if (const FLayerData* Data = LayerData.Find(LayerName))
+		{
+			LayerNames[Data->LayerIndex] = LayerName;
+			Layers[Data->LayerIndex] = Entry.Value->CreateLoadBalancingCalculator(OutCtx);
+		}
+	}
+
+	auto Calculator = MakeUnique<SpatialGDK::FLayerLoadBalancingCalculator>(LayerNames, MoveTemp(Layers));
+	OutCtx.Layers = Calculator.Get();
+
+	return Calculator;
+}
+
+bool ULayeredLBStrategy::IsStrategyWorkerAware() const
+{
+	for (auto Entry : LayerNameToLBStrategy)
+	{
+		if (!Entry.Value->IsStrategyWorkerAware())
+		{
+			return false;
+		}
+	}
+	return true;
+}
+
+SpatialGDK::FLoadBalancingDecorator* ULayeredLBStrategy::GetLoadBalancingDecorator() const
+{
+	if (!Decorator.IsValid())
+	{
+		if (!IsStrategyWorkerAware())
+		{
+			return nullptr;
+		}
+
+		TClassMap<uint32> GroupMap;
+		for (auto Entry : ClassPathToLayerName)
+		{
+			UClass* Class = Entry.Key.LoadSynchronous();
```

Comment:
Can't we use SoftClassPaths here to prevent the class load?

Reply:
I need the inheritance information. How would I be able to access it while using SoftClassPath?